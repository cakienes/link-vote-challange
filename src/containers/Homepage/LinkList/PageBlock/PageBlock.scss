.pageBlockHolder {
    text-align: center;
    font-weight: bold;
    .leftArrow,
    .rightArrow {
        font-size: 22px;
        display: inline-block;
        vertical-align: middle;
        margin: 0px 5px;
    }
    .numberBlock {
        display: inline-block;
        vertical-align: middle;
        font-size: 22px;
        .pageNumber {
            display: inline-block;

            height: 35px;
            width: 35px;

            &.selected {
                border: 1px solid #000;
            }
        }
    }
}
