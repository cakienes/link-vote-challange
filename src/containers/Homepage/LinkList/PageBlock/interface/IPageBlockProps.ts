export default interface IPageBlockProps {
    currentPage: number;
    totalSize: number;

    onClickPage: (page: number) => void;
}
