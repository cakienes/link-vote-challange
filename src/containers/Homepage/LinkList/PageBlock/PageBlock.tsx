import React from 'react';
import IPageBlockProps from './interface/IPageBlockProps';
import './PageBlock.scss';

export class PageBlock extends React.Component<IPageBlockProps, any> {
    render() {
        return (
            <div className="pageBlockHolder">
                <div className="leftArrow" onClick={this.onClickPrev}>{`<`}</div>
                <div className="numberBlock">{this.renderNumber()}</div>
                <div className="rightArrow" onClick={this.onClickNext}>{`>`}</div>
            </div>
        );
    }

    renderNumber = (): React.ReactNode => {
        const { totalSize, currentPage } = this.props;
        const numberArray: number[] = [];
        for (let i: number = 0; i <= totalSize; i++) {
            numberArray.push(i);
        }

        return numberArray.map((i: number) => {
            return (
                <div
                    className={`pageNumber ${i === currentPage ? 'selected' : undefined}`}
                    key={i}
                    onClick={() => this.props.onClickPage(i)}
                >
                    {i + 1}
                </div>
            );
        });
    };

    onClickPrev = (): void => {
        const { currentPage } = this.props;
        const prevPage = currentPage - 1;
        if (prevPage >= 0) this.props.onClickPage(prevPage);
    };

    onClickNext = (): void => {
        const { currentPage, totalSize } = this.props;
        if (totalSize !== currentPage) {
            this.props.onClickPage(currentPage + 1);
        }
    };
}

export default PageBlock;
