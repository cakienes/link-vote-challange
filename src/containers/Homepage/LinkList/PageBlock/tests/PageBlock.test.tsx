import { mount } from 'enzyme';
import React from 'react';
import IPageBlockProps from '../interface/IPageBlockProps';
import PageBlock from '../PageBlock';

describe('PageBlock.tsx', () => {
    const onClickPageMock = jest.fn();
    let pageBlockProps: IPageBlockProps = {
        currentPage: 0,
        onClickPage: onClickPageMock,
        totalSize: 2,
    };

    it('renders without crashing', async () => {
        const wrapper = mount(<PageBlock {...pageBlockProps} />);
        const instance: any = wrapper.instance();

        instance.renderNumber = jest.fn().mockReturnValue('');
        expect(instance.render()).toMatchSnapshot();
    });

    it('renders without crashing', async () => {
        const wrapper = mount(<PageBlock {...pageBlockProps} />);
        const instance: any = wrapper.instance();

        expect(instance.renderNumber()).toMatchSnapshot();
    });

    it('onClickPrev of LinkList while currentPage is 0', async () => {
        const wrapper = mount(<PageBlock {...pageBlockProps} />);
        const instance: any = wrapper.instance();

        await instance.onClickPrev();
        expect(onClickPageMock).not.toHaveBeenCalled();
    });

    it('onClickPrev of LinkList while currentPage is 2', async () => {
        pageBlockProps = {
            ...pageBlockProps,
            currentPage: 1,
        };
        const wrapper = mount(<PageBlock {...pageBlockProps} />);
        const instance: any = wrapper.instance();

        await instance.onClickPrev();
        expect(onClickPageMock).toHaveBeenCalledWith(0);
    });

    it('onClickNext of LinkList while totalSize is not equal currentPage', async () => {
        pageBlockProps = {
            ...pageBlockProps,
            currentPage: 1,
            totalSize: 2,
        };
        const wrapper = mount(<PageBlock {...pageBlockProps} />);
        const instance: any = wrapper.instance();

        await instance.onClickNext();
        expect(onClickPageMock).toHaveBeenCalledWith(2);
    });
});
