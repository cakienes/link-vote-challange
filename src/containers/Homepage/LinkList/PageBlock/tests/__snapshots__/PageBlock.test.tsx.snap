// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageBlock.tsx renders without crashing 1`] = `
<div
  className="pageBlockHolder"
>
  <div
    className="leftArrow"
    onClick={[Function]}
  >
    &lt;
  </div>
  <div
    className="numberBlock"
  >
    
  </div>
  <div
    className="rightArrow"
    onClick={[Function]}
  >
    &gt;
  </div>
</div>
`;

exports[`PageBlock.tsx renders without crashing 2`] = `
Array [
  <div
    className="pageNumber selected"
    onClick={[Function]}
  >
    1
  </div>,
  <div
    className="pageNumber undefined"
    onClick={[Function]}
  >
    2
  </div>,
  <div
    className="pageNumber undefined"
    onClick={[Function]}
  >
    3
  </div>,
]
`;
